### PR TITLE
[Draft] Add functionality to enable the usage of group style avatars in PersonaButtonCarousel

### PR DESCRIPTION
**Pull request by laminesm**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

There is a partner request to enable the usage of group style avatars in `PersonaButtonCarousel`. I added a boolean variable `isGroupStyle` in the `MSFPersonaButtonCarouselState` protocol. As the name suggests, this boolean is used to determine whether the carousel contains regular avatars or group avatars. 
I confirmed with design that there are no changes to the paddings (or any other values) for the group style carousel. 

Binary change:

| File | Before | After | Delta |
|------|--------|-------|-------|
| libFluentUI.a | 28,864,736 bytes | 28,880,512 bytes | 15,776 bytes |

### Verification

The changes were tested on the demo app.

| Light                                       | Dark                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot 2022-12-15 at 6 27 28 PM | Screenshot 2022-12-15 at 6 27 33 PM |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1456)

### CODE REVIEW

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/PersonaButtonCarouselDemoController.swift [hunk 1]
```diff
@@ -268,4 +279,16 @@ class PersonaButtonCarouselDemoController: DemoTableViewController {
         PersonaData(email: "carlos.slattery@contoso.com", subtitle: "Software Engineer")
     ]
 
+    private let groupPersonas: [PersonaData] = [
```
Comment (edjamesmsft) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/PersonaButtonCarouselDemoController.swift:282]: Can you not re-use the other persona data? I'm not sure I love the structure of this test data.

File: ios/FluentUI/PersonaButton/PersonaButton.swift [hunk 2]
```diff
@@ -61,8 +61,8 @@ public struct PersonaButton: View, TokenizedControlView {
     /// Creates a new `PersonaButton` instance.
     /// - Parameters:
     ///   - size: The` MSFPersonaButtonSize` value used by the `PersonaButton`.
-    public init(size: MSFPersonaButtonSize) {
-        let state = MSFPersonaButtonStateImpl(size: size)
+    public init(size: MSFPersonaButtonSize, isGroupStyle: Bool = false) {
```
Comment (edjamesmsft) [inline ios/FluentUI/PersonaButton/PersonaButton.swift:64]: Should this be a new convenience init instead?

Comment (edjamesmsft) [reply ios/FluentUI/PersonaButton/PersonaButton.swift:64]: (We shouldn't have breaking APIs if we can avoid it)

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/PersonaButtonCarouselDemoController.swift (hunk 1 above)
Comment (laminesm) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/PersonaButtonCarouselDemoController.swift:282]: It's possible, but I created this data to resemble the specs design shared with me. I've shared the files with you on teams.

File: ios/FluentUI/PersonaButton/PersonaButton.swift (hunk 2 above)
Comment (laminesm) [reply ios/FluentUI/PersonaButton/PersonaButton.swift:64]: Apparently you can't have convenience initializers in structs. 
Question: would this still be considered breaking changes even if the new parameter has a default value passed? It changes the API but it wouldn't break any of the existing code.

Comment (edjamesmsft) [reply ios/FluentUI/PersonaButton/PersonaButton.swift:64]: Try adding it to the objc demo and see if it breaks?

Comment (laminesm) [reply ios/FluentUI/PersonaButton/PersonaButton.swift:64]: PersonaButton doesn't seem to have objc exposure. Is there another way to add it to the objc demo?

**Pull request closed without merging**